### ISSUE TITLE
List SVN repository url and also check if URL is accessible

### ISSUE THREAD

**Issue by anantshri** [anantshri/svn-extractor#9]
SVN repository url should be checked if its accessible and it would be desirable if a basic password bruteforce could also be innitited for example

username = x then password as x or password

**Comment by ai-anant** [comment]
🤖 Watchdog analysis

**Issue:** "List SVN repository url and also check if URL is accessible" — feature request (2014).

## Root cause / current state

There is no bug here — this is a feature request, and part of it is **already implemented** in `svn_extractor.py`:

- **URL accessibility check already exists**: `main()` performs a connectivity check on the target before doing anything else (`svn_extractor.py:203-211`): it GETs the URL, treats status codes 200/403/500 as "URL is active", prints "Problem connecting to URL" / "URL returns <code>" and exits otherwise.
- **What's missing**:
  1. **"List SVN repository url"** — the tool only accepts a *single* URL (`--url`, `svn_extractor.py:164`) and never prints the actual SVN repository URL it discovers. For SVN ≥ 1.7 working copies, `readwc()` (`svn_extractor.py:63-97`) reads `wc.db` but only queries the `NODES` table (file list + `changed_author`); it never reads the **`REPOSITORY` table** (`root` column), which contains the canonical repository URL (`select root from repository;`). For SVN < 1.6, the repository URL is present in `.svn/entries` (the line after the format version / `dir` marker) but `readsvn()` (`svn_extractor.py:20-60`) also ignores it. So the tool never surfaces *which* repository it is hitting.
  2. **Batch checking** — no way to feed a list of URLs and get an accessibility report per URL.
  3. **Auth probing** — the "basic password bruteforce" (try `username:username`, `username:password`) does not exist in any form.

## Existing PR status

No open or merged PR references issue #9. Open PRs are unrelated: #20 (custom output directory), #19 (Sourcery refactor), #17 (shebang). The tool is now Python 3 (PR #18 merged), so any new code should target Python 3.

## Fix guidance

The cleanest, minimal implementation, all inside `svn_extractor.py`:

1. **Report the repository URL** (fulfils "List SVN repository url" literally):
   - In `readwc()`: after opening `wc.db`, add `c.execute('select root from repository;')` and print the result (guard with try/except — the `REPOSITORY` table exists in all SVN 1.7+ wc.db schemas).
   - In `readsvn()` (`.svn/entries`, pre-1.7): the repository URL is the line immediately following the first `dir` line of the entries file — capture and print it when parsing.

2. **Batch accessibility check mode** (fulfils "check if URL is accessible" for many URLs). Add an optional `--urllist FILE` argument (one URL per line), and in `main()` before the single-URL logic:

   ```python
   if x.urllist:
       with open(x.urllist) as f:
           urls = [l.strip() for l in f if l.strip()]
       for u in urls:
           try:
               r = requests.get(u, verify=False, timeout=10, proxies=proxy_dict)
               active = "ACTIVE" if r.status_code in (200, 403, 500) else "inactive"
               print("{} : {} ({})".format(u, r.status_code, active))
           except Exception as e:
               print("{} : ERROR ({})".format(u, e))
       exit()
   ```

   Keep the same 200/403/500 heuristic already used at `svn_extractor.py:212` so behaviour is consistent. Note the existing check has no `timeout=` — adding `timeout=10` avoids hangs on unreachable hosts.

3. **Optional auth probe** (`--auth-probe`): only after the URL is confirmed active, attempt GETs with `requests.get(url, auth=(username, username))` and `auth=(username, password)` for each username found via `--userlist` (or a `--user` argument). Make it strictly opt-in and print a prominent banner that probing is only legal against systems you are authorized to test. Given the sensitivity of credential-guessing functionality, this part is best implemented deliberately by the maintainer rather than bolted on blindly.

**Pitfalls:**
- `--userlist` data is only collected when a wc.db/entries parse *succeeds* (`svn_extractor.py:226,241`); the auth probe must handle the case where no usernames were discovered.
- Keep the probe to the two default combos described in the issue; do not add a full dictionary attack.
- `requests.packages.urllib3.disable_warnings(InsecureRequestWarning)` (`svn_extractor.py:11-12`) is removed in newer `requests` versions — while you're in the file, switch to `import urllib3; urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)` to stay compatible with requests ≥ 2.32.

## PR status

No new PR opened. This is a 12-year-old feature request with open design questions (input format, output format, scope/opt-in of the auth probe) — not a quick, unambiguous bug fix, so it gets guidance only. The maintainer can decide which parts to adopt.

## Verification

- `python3 -m py_compile svn_extractor.py` passes on current `master` (verified in this analysis).
- After implementing: run against a test wc.db (`sqlite3 wc.db 'select root from repository;'` to confirm expected URL) and against a reachable/unreachable URL list to confirm the `--urllist` report.
